hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataWriteUtils.java

```diff
@@ -111,6 +112,10 @@ public static HoodieWriteConfig createMetadataWriteConfig(
             // deltacommits having corresponding completed commits. Therefore, we need to compact all fileslices of all
             // partitions together requiring UnBoundedCompactionStrategy.
             .withCompactionStrategy(new UnBoundedCompactionStrategy())
+            // Check if log compaction is enabled, this is needed for tables with lot of records.
+            .withLogCompactionEnabled(writeConfig.isLogCompactionEnabled())
+            // This config is only used if enableLogCompactionForMetadata is set.
```

Comment:
not sure I get your comment here "This config is only used if enableLogCompactionForMetadata is set". from the code, it looks like we fetch from  writeConfig.isLogCompactionEnabled().

Reply:
i think it's a typo. i didn't find any method for `enableLogCompactionForMetadata`

Reply:
Fixed the comment.

Reply:
Yeah, log compaction for MDT and DT should not share the same config option I think.